src/lib/routes/admin-api/api-token.ts

```diff
@@ -419,23 +392,33 @@ export class ApiTokenController extends Controller {
         if (user.isAPI && user.permissions.includes(ADMIN)) {
             return allTokens;
         }
+
         const userPermissions =
             await this.accessService.getPermissionsForUser(user);
 
-        const allowedTokenTypes = [
-            ADMIN,
-            READ_CLIENT_API_TOKEN,
-            READ_FRONTEND_API_TOKEN,
-        ]
-            .filter((readPerm) =>
-                userPermissions.some(
-                    (p) => p.permission === readPerm || p.permission === ADMIN,
-                ),
+        const readTokenFns = userPermissions
+            .filter((p) =>
+                [
+                    ADMIN,
+                    READ_CLIENT_API_TOKEN,
+                    READ_FRONTEND_API_TOKEN,
+                ].includes(p.permission),
             )
-            .map(permissionToTokenType)
-            .filter((t) => t);
-        return allTokens.filter((token) =>
-            allowedTokenTypes.includes(token.type),
+            .map(
+                ({ permission }) =>
+                    (type: ApiTokenType) =>
+                        permission === ADMIN ||
+                        (type === ApiTokenType.FRONTEND &&
+                            permission === READ_FRONTEND_API_TOKEN) ||
+                        ([ApiTokenType.CLIENT, ApiTokenType.BACKEND].includes(
+                            type,
+                        ) &&
+                            permission === READ_CLIENT_API_TOKEN),
+            );
+
+        const accessibleTokens = allTokens.filter((token) =>
+            readTokenFns.some((canRead) => canRead(token.type)),
```

Comment:
Personally I don't find this much more readable than before 😕 Something tells me we may be able to do better.